desc/grid.py

```diff
@@ -331,6 +316,33 @@ def node_pattern(self):
         """str: Pattern for placement of nodes in (rho,theta,zeta)."""
         return self.__dict__.setdefault("_node_pattern", "custom")
 
+    @property
+    def nodes(self):
+        """ndarray: Node coordinates, in (rho,theta,zeta)."""
+        return self.__dict__.setdefault("_nodes", np.array([]).reshape((0, 3)))
+
+    @nodes.setter
```

Comment:
1. remove the setter methods so that they are not publicly writable
2. set the read only attribute on the numpy arrays to make them immutable
3. recast to jax array

Is your suggestion to do just number 1? Or if you want to to 1&2, then why not do option 3 to avoid the need for this: https://github.com/PlasmaControl/DESC/blob/6fad1a8158b4ed3aed3f75e6dc2d495041f62a06/desc/compute/utils.py#L1209

Reply:
I was just thinking about #1